department-of-veterans-affairs/vets-api
# Add additional logs to Pension submission

## Summary

- Adds additional logs to the `ClaimDocumentsController` to track attempted, successful, and failed document uploads
- Adds additional details to the `ClaimsBaseController` to track user ID and claim ID where available

## Related issue(s)
- [department-of-veterans-affairs/va.gov-team#64938](https://github.com/department-of-veterans-affairs/va.gov-team/issues/64938)
- continues work begun in [department-of-veterans-affairs/vets-api/pull/14467](https://github.com/department-of-veterans-affairs/vets-api/pull/14467)

## Testing done

- Run `tail -f log/development.log`, the API and the website
- Enable http://localhost:3000/flipper/features/pension_form_enabled
- Login and begin an application at http://localhost:3001/pension/application/527EZ/
- When you get to [the document upload page](http://localhost:3001/pension/application/527EZ/documents), upload a valid file. 
     - You should see a log of the attempt in `log/development.log`
     - You should see a log that the attempt succeeded in `log/development.log`, and it should include the attachment ID
- Attempt to upload a locked PDF and let it fail
     - You should see a log of the attempt in `log/development.log`
     - You should see a log that the attempt failed in `log/development.log`
- Delete the failed document upload and submit the claim
     - You should see a log of the attempt, including user ID, in `log/development.log`
     - You should see a log that the attempt succeeded, including user ID, in `log/development.log`

## Screenshots
<img width="939" alt="attempt_attachment" src="https://github.com/department-of-veterans-affairs/vets-api/assets/21046714/04bbbe56-ec80-4bd9-8a50-4ab185083bbf">
<img width="878" alt="success_attachment" src="https://github.com/department-of-veterans-affairs/vets-api/assets/21046714/80bc1ddc-a059-443e-9ee2-85a6c4302157">
<img width="950" alt="error_attachment" src="https://github.com/department-of-veterans-affairs/vets-api/assets/21046714/fec2187a-d718-44f6-a678-3300bc55599f">
<img width="983" alt="begin_claim" src="https://github.com/department-of-veterans-affairs/vets-api/assets/21046714/43dcaa57-f120-4101-b1c6-06fddd6436cd">
<img width="734" alt="submitted_claim" src="https://github.com/department-of-veterans-affairs/vets-api/assets/21046714/e8b41add-aba1-4d54-9926-eaa995f2b6df">

## What areas of the site does it impact?
`ClaimDocumentsController` and `ClaimsBaseController` submissions

## Acceptance criteria


## Requested Feedback

There may be issues submitting Pension claims due to recent changes in the UI; if so, please temporarily revert to an earlier commit of the website.
